isis/src/base/apps/trackextract/trackextract.cpp

```diff
@@ -0,0 +1,213 @@
+    msg += " If you want to create a tracking cube, run a mosaic program.";
+    throw IException(IException::Programmer, msg, _FILEINFO_);
+  }
+}
+
+
+/**
+ * Creates the mosaic cube by copying the input cube without the tracking band.
+ * Then, removes the tracking table from the mosaic label and adds a group pointing
+ * to the tracking cube.
+ *
+ * @param ui    The user interface
+ * @param bands The indices of the bands that are not the tracking band
+ */
+void createMosaicCube(UserInterface ui, QString bands) {
+  ProcessByLine p;
+  CubeAttributeInput inAtt = CubeAttributeInput("+" + bands);
+  p.SetInputCube(ui.GetFileName("FROM"), inAtt);
+  p.SetOutputCube("TO");
+  p.StartProcess(copyPixels);
+  p.EndProcess();
+
+  Cube mosaicCube;
+  try {
+    mosaicCube.open(ui.GetFileName("TO"),"rw");
+  }
+  catch (IException &e) {
+    throw IException(IException::User,
+                     "Unable to open the file [" + ui.GetFileName("TO") + "] as a cube.",
+                     _FILEINFO_);
+  }
+  Pvl *mosaicLabel = mosaicCube.label();
+
+  if (mosaicLabel->hasObject("Table")) {
+    PvlObject &tableObject = mosaicLabel->findObject("Table");
+    if (tableObject.hasKeyword("Name")) {
+      PvlKeyword &tableName = tableObject.findKeyword("Name", Pvl::Traverse);
+      if (tableName[0] == "InputImages") {
+        mosaicLabel->deleteObject("Table");
+      }
+    }
+  }
+
+  // Add Tracking Group to the mosaic cube
+  PvlGroup trackingGroup = PvlGroup("Tracking");
+  PvlKeyword trackingName = PvlKeyword("Filename");
+  FileName cubeName = FileName(ui.GetFileName("TO"));
+  trackingName.setValue(cubeName.baseName() + "_tracking.cub"); //Strip off path and add _tracking
+  trackingGroup.addKeyword(trackingName);
+  mosaicLabel->addGroup(trackingGroup);
+
+  mosaicCube.close();
+}
+
+
+/**
+ * Creates the tracking cube by copying the input cube with only the tracking band.
+ * Then, goes through each pixel and subtracts the input cube's pixel type's min value
+ * because ProcesMosaic used to add the min value to each pixel.
+ *
+ * @param ui           The user interface
+ * @param trackingBand The index of the tracking band
+ */
+void createTrackCube(UserInterface ui, QString trackBand) {
+  ProcessByLine p;
+
+  CubeAttributeInput inAtt = CubeAttributeInput("+" + trackBand);
+  p.SetInputCube(ui.GetFileName("FROM"), inAtt);
+
+  FileName cubeName = FileName(ui.GetFileName("TO"));
+  // Strip off any extensions and add _tracking
+  QString trackingName = cubeName.path() + "/" + cubeName.baseName() + "_tracking.cub";
+  Cube inputCube = Cube(ui.GetFileName("FROM"));
+  int numSample = inputCube.sampleCount();
+  int numLine = inputCube.lineCount();
+
+  QString outAttString = "+UNSIGNEDINTEGER+" + QString::number(VALID_MINUI4) + ":";
+  outAttString += QString::number(VALID_MAXUI4);
+  CubeAttributeOutput outAtt = CubeAttributeOutput(outAttString);
```

Comment:
You can create a CubeAttributeOutput object and then set the pixel type, min, and max through its own methods instead of by using the UI string.